src/app/modules/app/pages/programs/programs.component.ts

```diff
@@ -11,29 +14,38 @@ import { ProgramService } from 'src/app/shared/services/program.service';
 export class ProgramsComponent {
   @ViewChild('sidenav', { static: false }) sidenav: MatSidenav | undefined;
   isSidenavOpen = true;
+  isLoading = true;
 
   headers = ['id', 'name', 'description'];
   data: Program[] | Program | null | any = [];
 
-  constructor(public programService: ProgramService) {
+  constructor(public programService: ProgramService, private snackBar: MatSnackBar) {
     this.fetchData();
   }
 
   fetchData() {
-    try {
-      this.programService.getAll().subscribe(
-        (data: Program[] | Program | null) => {
-          this.data = data;
-        },
-        error => {
-          console.log(
-            'Something went wrong while trying to retrieve data from API',
-            error
-          );
-        }
-      );
-    } catch (error) {
-      console.log(error);
-    }
+    this.isLoading = true;
+    this.programService.getAll().subscribe({
+      next: this.handleSuccess.bind(this),
+      error: this.handleError.bind(this),
+    });
+  }
+  private handleSuccess(data: Program[]): void {
+    this.data = data;
+    this.isLoading = false;
+  }
```

Comment:
Quizá deberíamos evitar el uso de la palabra "datos". Todo puede ser "datos".
Aquí "datos" podría ser programList, o listOfPrograms, o programas, o algo similar

Reply:
Done, changed variable name to programList

Also, changed the FetchData function name

![imagen](https://github.com/JU-DEV-LatamBootcamp/Klockan-Frontend/assets/151577414/bd11adfb-c878-4ace-842a-8d16437672a5)